### PR TITLE
[RA1 Ch08] Add gaps in openstack release ocata, pike and stein

### LINKED ISSUE #842: [RA1 Ch08] Document the Openstack release differences
Issue aims to identify differences among Openstack Ocata, Pike and Stein as discussed.

### PR DESCRIPTION
**Pull request by deebhatia**
* Add API versions
* Add features of pike and stein
* Add upgrade check for Stein

Fixes #842

### CODE REVIEW

**Comment by collivier** [comment]
I don't see the point to compare the OpenStack releases here.

RA1 API sections describes minimal microversions and then should match from Pike to Stein (it seems ok when testing Functest SUTs already compliant in both versions).

I think here we may list here instead where RI is not compliant. Apart a few features (at least 1 Cinder feature) introduced in Pike, it's mostly failing due to missing mainline supports (live migration, resize server, etc).. 
https://build.opnfv.org/ci/job/airship-latest-daily/81/
https://build.opnfv.org/ci/job/cntt-latest-daily/27/

**Comment by deebhatia** [comment]
@collivier So, where should we define the release differences? RC is not a good place for defining this. I can move them in RA. 
To list that failure in Cinder, we should use `RC Ch11` as it is a capability failure captured in CI.

**Comment by pgoyal01** [comment]
@deebhatia Stein is not yet a version that has been agreed to by CNTT. Whatever is the next version (say Stein) specified by CNTT, hopefully in Prague, A new version of some of the chapters for RA-1 will have to be created with the differences between Pike and Stein.

**Comment by deebhatia** [comment]
@pgoyal01 That's okay if it is not finalised yet. This just serve as a reference to what Stein has over Pike and will serve as an assist in Prague. I can add Queens in additional if you want. Adding all of the versions might be an overkill. See partial discussion [here](https://github.com/cntt-n/CNTT/issues/419).

**Comment by collivier** [comment]
@pgoyal01 @deebhatia For my understanding, it's preparing indirectly the next CNTT updates after January (part of the testing is ready too). I don't see why we should go to the past here.

**Comment by deebhatia** [comment]
@collivier you mean Ocata? AFAIK, Ocata is there because RI is using that version. What release airship is moving forward with upcoming ? I see references of Stein [here](https://git.opnfv.org/airship/tree/type/cntt/software/config/versions.yaml).

File: doc/ref_impl/cntt-ri/chapters/chapter08.md [hunk 1]
```diff
@@ -10,9 +10,27 @@
...
-## 8.2 Heading
+## 8.2 OpenStack Release Gaps
+Section contains the key differences between the chosen baseline version (Pike) and chosen version for RI (Ocata).
+Along with those two versions, it also documents the differences of Pike and Stein. The table below gives only an
+overview of the differences. For detailed changes, please check the
+[https://releases.openstack.org](https://releases.openstack.org).
+
+| Service Name | OpenStack (Ocata) (RI version) | OpenStack (Pike) (CNTT baseline) | OpenStack (Stein) |
+| --- | --- | --- | --- |
+| Keystone | 3.8 | 3.8 | 3.12<br>- Support for project tags, application credential, domain level resource limits, JWTs.<br> - Introduced system scoped roles<br> - Introduced new role 'reader' along with 'member' and 'admin' |
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri/chapters/chapter08.md]: JWTs -- first use acronym please expand

**Comment by mf4716** [comment]
@deebhatia - can you please review/resolve open conversation items?  Thanks.

File: doc/ref_impl/cntt-ri/chapters/chapter08.md (hunk 1 above)
Comment (deebhatia) [reply doc/ref_impl/cntt-ri/chapters/chapter08.md]: Removed the acronym.

**Comment by rabiabdel** [comment]
@deebhatia i suggest moving this content to RA1 Ch08 where it is talking about GAPS. is that OK?

**Comment by deebhatia** [comment]
> @deebhatia i suggest moving this content to RA1 Ch08 where it is talking about GAPS. is that OK? GAPS here is meant to be around GAPS in installers and eco-systems.

@rabi-abdel Agree. Moved.

**Comment by rabiabdel** [comment]
> > @deebhatia i suggest moving this content to RA1 Ch08 where it is talking about GAPS. is that OK? GAPS here is meant to be around GAPS in installers and eco-systems.
> 
> @rabiabdel @rabi-abdel Agree. Moved.

@deebhatia OK you need to teach me how did you do it that quick? you have magic tools or something :)

**Comment by rabiabdel** [comment]
it will be good to have one more approval, if we dont get this approval before Snezka release, i will force merge it.

**Pull request merged**